### DISCUSSION TITLE
How to compat with binary files compatible with `riscv-none-embed`

### DISCUSSION THREAD

**Discussion by dreamcmi** [xpack-dev-tools/riscv-none-elf-gcc-xpack#12] [Q&A] [answered]
Ready to upgrade the company's GCC to version 12. The previous use of the `riscv-none-embed-gcc-xpack` repo. The binary files generated by it are beginning of `riscv-none-embed`, but this repo is `riscv-none-elf` . If we want to be compatible with the old version through the file name, which files need to be modified to realize such demand?

**Comment by ilg-ul** [comment, marked as answer]
The old names prefixed with `riscv-none-embed-` are deprecated and **should not** be used for future projects, since there will be no further releases with these name.

I don't know your tools and your environment, but I don't think you can use your old projects unchanged, you have to edit them to update the file names, to the new `riscv-none-elf-` prefix.

**Reply by dreamcmi** [reply]
I personally think that the prefix can be modified, but the projects in the company are very complex and numerous due to historical reasons, so the leader needs a softer transition method. This part of the modification does not need to be committed to the xpack repo, but provides a modification. The ideas are easy for developers to use independently.

**Comment by dreamcmi** [comment]
I tried it myself and found that it is feasible. The following is the repo I modified. It is not easy to make it, because I found that there is no patch for newlib in `xbb-helper-xpack`. I hope it can be fixed in the next version. , although there is no patch to be applied, the relevant sh of `gcc` `gdb` `binutils` all retain the function input parameters that the patch will be applied to.
like this https://github.com/xpack-dev-tools/xbb-helper-xpack/blob/xpack/dependencies/gcc.sh#L114
but the newlib is https://github.com/xpack-dev-tools/xbb-helper-xpack/blob/xpack/dependencies/newlib-cross.sh#L72

My modified version is here https://github.com/dreamcmi/riscv-none-elf-gcc-xpack/commit/ec36dca8e1c9203e2270302d2513a1e9a69b8112

**Comment by ilg-ul** [comment]
Not everything that can be done, should be done. 

There will be no next release of `riscv-none-embed-gcc`, that project was deprecated.

Please use `riscv-none-elf-gcc`, otherwise you're entirely on your own.

**Reply by dreamcmi** [reply]
Yes. I agree with you. Thanks.
